src/AstGen.zig

```diff
@@ -7897,20 +7908,24 @@ fn callExpr(
     call: Ast.full.Call,
 ) InnerError!Zir.Inst.Ref {
     const astgen = gz.astgen;
-    const lhs = try expr(gz, scope, .none, call.ast.fn_expr);
+    const tree = astgen.tree;
+
+    // If the lhs is a field access, we need to generate a special field_call_bind
+    // instruction first.  If this is a inst.func() call, this instruction will capture
+    // the value of the first parameter before evaluating the other parameters.
+    // TODO NO_COMMIT make sure .ref is correct here
```

Comment:
`ResultLoc.ref` means that you're going to get back a pointer to `call.ast.fn_expr` no matter what. For example if the fn expr was an identifier to a local, you'll get the `alloc` instruction, rather than a `load` instruction of that alloc. If the fn expr is an expression, such as `(a + b)` then the result of that expression will be stored in the stack and you'll get a pointer to the stack. The `ref` instruction does all that - stuffs an arbitrary value into an alloc then returns the alloc.